src/controllers/authenticationController.ts

```diff
@@ -200,6 +207,115 @@ export default class AuthenticationController {
       sessionTrackingId
     );
 
+    const errorOrMaybeAssertionRef = this.lollipopParams.isLollipopEnabled
+      ? await this.sessionStorage.getLollipopAssertionRefForUser(user)
+      : E.right(O.none);
+
+    if (E.isLeft(errorOrMaybeAssertionRef)) {
+      return ResponseErrorInternal(
+        "Error retrieving previous lollipop configuration"
+      );
+    }
+
+    if (
+      this.lollipopParams.isLollipopEnabled &&
+      O.isSome(errorOrMaybeAssertionRef.right)
+    ) {
+      // Sending a revoke message for previous assertionRef related to the same fiscalCode
+      // This operation is fire and forget
+      this.lollipopParams.lollipopService
+        .revokePreviousAssertionRef(errorOrMaybeAssertionRef.right.value)
+        .catch(err => {
+          log.error(
+            "acs: error sending revoke message for previous assertionRef [%s]",
+            err
+          );
+        });
+    }
+
+    const errorOrActivatedPubKey = await pipe(
+      // Delete the reference to CF and assertionRef for lollipop.
+      // This operation must be performed even if the lollipop FF is disabled
+      // to avoid inconsistency on CF-key relation if the FF will be re-enabled.
+      TE.tryCatch(
+        () =>
+          this.sessionStorage.delLollipopAssertionRefForUser(
+            spidUser.fiscalNumber
+          ),
+        E.toError
+      ),
+      TE.chainEitherK(identity),
+      TE.chain(
+        TE.fromPredicate(
+          _ => _ === true,
+          () => new Error("Error on LolliPoP initialization")
+        )
+      ),
```

Comment:
```suggestion
      TE.chain(
        TE.fromPredicate(
          delLollipopAssertionRefResult => delLollipopAssertionRefResult === true,
          () => new Error("Error on LolliPoP initialization")
        )
      ),
```
or in alternative we can try `TE.filterOrElse`

Reply:
fixed in 7df56b4